internal/commands/test.go

```diff
@@ -423,3 +418,79 @@ func buildRego(trace bool, query string, input interface{}, compiler *ast.Compil
 
 	return regoObj, buf
 }
+
+func parseFileList(fileList []string, input string) ([]string, error) {
+	var files []string
+	for _, file := range fileList {
+		if file == "" {
+			continue
+		}
+
+		if file == "-" {
+			files = append(files, "-")
+			continue
+		}
+
+		fileInfo, err := os.Stat(file)
+		if err != nil {
+			return nil, fmt.Errorf("get file info: %w", err)
+		}
+
+		if fileInfo.IsDir() {
+			directoryFiles, err := getFilesFromDirectory(file, input)
+			if err != nil {
+				return nil, fmt.Errorf("get files from directory: %w", err)
+			}
+
+			files = append(files, directoryFiles...)
+		} else {
+			files = append(files, file)
+		}
+	}
+
+	if len(files) == 0 {
+		return nil, fmt.Errorf("no files found")
+	}
+
+	return files, nil
+}
+
+func getFilesFromDirectory(directory string, input string) ([]string, error) {
+	var files []string
+	err := filepath.Walk(directory, func(currentPath string, info os.FileInfo, err error) error {
+		if err != nil {
+			return fmt.Errorf("walk path: %w", err)
+		}
+
+		if info.IsDir() {
+			return nil
+		}
+
+		if input == "" {
+			for _, input := range parser.ValidInputs() {
```

Comment:
Small nitpick: Overriding input here makes it slightly hard to read.

Reply:
Agreed.. I know I fixed this at some point, must not have pushed

Reply:
Ah right, I ended up just removing that piece. Since this is brand new functionality, and that I feel `input` is starting to cause too much branching logic, I just took the stance of "using a directory is short hand for typing out every supported file", and not change the behavior of `input` at all.

Later I want to revisit how `input` is used, but how I have it pushed now, keeps the behavior more or less the same.